data-pipeline.yml

```diff
@@ -0,0 +1,211 @@
+version: 1
+tasks:
+  - ID: commit-retrieval
+    created: {$fromNow: ''}
+    deadline: {$fromNow: '12 hours'}
+    provisionerId: aws-provisioner-v1
+    workerType: release-svc-compute
+    payload:
+      maxRunTime: 86400
+      image: mozilla/bugbug-commit-retrieval
+
+      artifacts:
+        public/commits.json.xz:
+          path: /data/commits.json.xz
+          type: file
+
+    metadata:
+      name: bugbug commit retrieval
+      description: bugbug commit retrieval
+      owner: release-mgmt-analysis@mozilla.com
+      source: https://github.com/mozilla/bugbug/raw/master/data-pipeline.yml
+
+  - ID: bugs-retrieval
+    created: {$fromNow: ''}
+    deadline: {$fromNow: '1 hour'}
+    provisionerId: aws-provisioner-v1
+    workerType: releng-svc
+    payload:
+      maxRunTime: 86400
+      image: mozilla/bugbug-bugs-retrieval
+
+      artifacts:
+        public/bugs.json.xz:
+          path: /data/bugs.json.xz
+          type: file
+
+      features:
+        taskclusterProxy:
+          true
+
+    metadata:
+      name: bugbug bugs retrieval
+      description: bugbug bugs retrieval
+      owner: release-mgmt-analysis@mozilla.com
+      source: https://github.com/mozilla/bugbug/raw/master/data-pipeline.yml
+
+  - ID: train-component
+    created: {$fromNow: ''}
+    deadline: {$fromNow: '1 hour'}
+    provisionerId: aws-provisioner-v1
+    workerType: releng-svc-compute
+    dependencies:
+      - commit-retrieval
+      - bugs-retrieval
+    payload:
+      maxRunTime: 86400
+      image: mozilla/bugbug-train-component
+
+      artifacts:
+        public/componentmodel.xz:
+          path: /componentmodel.xz
+          type: file
+
+    metadata:
+      name: bugbug train component model
+      description: bugbug train component model
+      owner: release-mgmt-analysis@mozilla.com
+      source: https://github.com/mozilla/bugbug/raw/master/data-pipeline.yml
+
+  - ID: train-defect
+    created: {$fromNow: ''}
+    deadline: {$fromNow: '1 hour'}
+    provisionerId: aws-provisioner-v1
+    workerType: releng-svc-compute
+    dependencies:
+      - commit-retrieval
+      - bugs-retrieval
+    payload:
+      maxRunTime: 86400
+      image: mozilla/bugbug-train-defect
+
+      artifacts:
+        public/defectenhancementtaskmodel.xz:
+          path: /defectenhancementtaskmodel.xz
+          type: file
+
+      features:
+        taskclusterProxy:
+          true
```

Comment:
I think all train task won't need access to secrets

Reply:
Is it only for accessing secrets? Would they need it for pushing / getting artifacts?

Reply:
It is needed for a few things that require auth, but artifacts do not require it IIRC.